src/main/java/ch/njol/skript/expressions/ExprExplodedBlocks.java

```diff
@@ -33,37 +41,137 @@
 import ch.njol.skript.doc.Since;
 import ch.njol.skript.lang.Expression;
 import ch.njol.skript.lang.ExpressionType;
-import ch.njol.skript.lang.SkriptParser;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
 import ch.njol.skript.lang.util.SimpleExpression;
 import ch.njol.util.Kleenean;
 
 @Name("Exploded Blocks")
 @Description("Get all the blocks that were destroyed in an explode event")
-@Examples({"on explode:",
+@Examples({
+	"on block explode:",
+	"\tset {_exploded-blocks::*} to exploded blocks",
+	"\t",
 	"\tloop exploded blocks:",
-	"\t\tadd loop-block to {exploded::blocks::*}"})
+	"\t\tif loop-block is sand:",
+	"\t\t\tset loop-block to stone",
+	"\t",
+	"\tset exploded blocks to blocks in radius 3 around event-location  # this will clear exploded blocks and set it to the specified blocks",
+	"\tremove all chests from exploded blocks",
+	"\tremove 3 stone from exploded blocks # will remove only 3 stone blocks",
+	"\tremove stone from exploded blocks # will remove all stones blocks - 'any' and 'all' will do the same",
+	"\tadd blocks in radius 3 around event-location to exploded blocks",
+	"\tdelete exploded blocks # will clear exploded blocks",
+	"\t",
+	"on entity explode:",
+	"\tremove all stone from exploded blocks",
+	"\t",
+	"on explode:",
+	"\tremove all stone from exploded blocks"})
 @Events("explode")
-@Since("2.5")
+@Since("2.5, INSERT VERSION (changing the list)")
 public class ExprExplodedBlocks extends SimpleExpression<Block> {
 
 	static {
-		Skript.registerExpression(ExprExplodedBlocks.class, Block.class, ExpressionType.COMBINED, "[the] exploded blocks");
+		Skript.registerExpression(ExprExplodedBlocks.class, Block.class, ExpressionType.SIMPLE, "[the] exploded blocks");
 	}
 	
 	@Override
-	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, SkriptParser.ParseResult parseResult) {
-		if (!getParser().isCurrentEvent(EntityExplodeEvent.class)) {
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		if (!getParser().isCurrentEvent(BlockExplodeEvent.class, EntityExplodeEvent.class)) {
 			Skript.error("Exploded blocks can only be retrieved from an explode event.");
 			return false;
 		}
 		return true;
 	}
 	
-	@Nullable
 	@Override
+	@Nullable
 	protected Block[] get(Event e) {
-		List<Block> blockList = ((EntityExplodeEvent) e).blockList();
-		return blockList.toArray(new Block[blockList.size()]);
+		return getBlocks(e).toArray(new Block[0]);
+	}
+
+	@Override
+	@Nullable
+	public Iterator<? extends Block> iterator(Event e) {
+		List<Block> blocks = getBlocks(e);
+		return new ArrayList(blocks).iterator(); // Prevent ConcurrentModificationException
+	}
+
+	@Override
+	public @Nullable <R> Expression<? extends R> getConvertedExpression(Class<R>... to) {
```

Comment:
What's the reason behind this method? If there is one, write a comment in the code for, otherwise remove it